File: ropchain/gadgets/asm/add.py
from ropchain.gadgets import gadget, util
from ropchain import ropchain

#TODO seprate add reg, reg; add reg, imm
def find(op1, op2, gadgets, canUse):
    rop, canUse = gadget.find(gadgets, canUse, 'add', op1, op2)
    return util.optROPChain(rop)

# def fromInc(op1, gadgets, canUse):
#     _inc = inc.find(op1, gadgets, canUse)

File: ropchain/lib/src/asm/add.cpp
#include "add.h"

OptROP Add::find(const Operand& op1, const Operand& op2,
        const Gadgets& gadgets, RegSet& aval) {
    auto gadget = Util::find(gadgets, aval, "add", op1, op2);
    return Util::toOptROP(gadget);
}
